src/Utility/ExprFunctions.C

```diff
@@ -342,6 +342,9 @@ Real EvalFunction::deriv (const Vec3& X, int dir) const
   }
   else if (!IAmConstant)
   {
+    if (gradient[--dir])
+      return gradient[dir]->evaluate(X);
```

Comment:
This will crash if someone accidentally invokes this method with dir > 4 (unlike before). So better replace by the equivalent, but safer
```
  if (gradient[3])
    return gradient[3]->evaluate(X);
```

Reply:
indeed.